### ISSUE TITLE
[gui error report] Error: Initializing daemon...

### ISSUE THREAD

**Issue by jc-frosty** [ipfs/ipfs-desktop#2884]
I have been having issues connecting to the daemon via the webui (browser) and have tried various steps towards trying to get a working.  IPFS has been broken on the two machines that I have installed it over different versions of both IPFS and MacOS but starting with the same issue, not being able to connect to the daemon.  I have tried to reinstall the desktop in addition to trying to install just the cli Kubo client, both had their own failures(the desktop didn't fix the issue, and the cli started with being the wrong version, which was fixed by IPFS-update(plagued by permission issues), but lead to an issue about a lock file(probably caused by an unkillable IPFS daemon(kill all, and kill 'PID' didn't work, it just restarted immediately as evidenced by a new PID).  Only after deleting both lock files was I able to kill it.  In an attempt to get a usable IPFS daemon, I compiled the code for a desktop daemon on my machine but that lead to this error.  As of this error, there is no IPFS daemon in shown in activity monitor.

- **Desktop**: 0.39.0
- **OS**: darwin 23.6.0 x64
- **Electron**: 32.2.0
- **Chrome**: 128.0.6613.178
- **Hardware**: Intel

```
Error: Initializing daemon...
Kubo version: 0.31.0
Repo version: 16
System version: amd64/darwin
Golang version: go1.23.2

Error: open /Users/frosty/.ipfs/datastore/LOCK: permission denied

    at errorTemplate (/Users/frosty/Downloads/Projects/ipfs-desktop/src/daemon/migration-prompt.js:78:49)
    at loadWindow (/Users/frosty/Downloads/Projects/ipfs-desktop/src/daemon/migration-prompt.js:129:26)
    at showMigrationPrompt (/Users/frosty/Downloads/Projects/ipfs-desktop/src/daemon/migration-prompt.js:134:3)
    at startIpfsWithLogs (/Users/frosty/Downloads/Projects/ipfs-desktop/src/daemon/daemon.js:193:9)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async startDaemon (/Users/frosty/Downloads/Projects/ipfs-desktop/src/daemon/daemon.js:216:27)
    at async startIpfs (/Users/frosty/Downloads/Projects/ipfs-desktop/src/daemon/index.js:49:17)
    at async setupDaemon (/Users/frosty/Downloads/Projects/ipfs-desktop/src/daemon/index.js:116:3)
    at async Promise.all (index 1)
    at async run (/Users/frosty/Downloads/Projects/ipfs-desktop/src/index.js:68:5)
```

**Comment by lidel** [comment]
kubo cannot open `~/.ipfs/datastore/LOCK` because that file is not owned by your user. It is the usual result of running `ipfs` or `ipfs-update` under `sudo`, which leaves parts of `~/.ipfs` owned by root.

Give the repo back to your user, then start IPFS Desktop again:

```console
$ sudo chown -R "$(whoami)" ~/.ipfs
```

Closing. If it still fails after that on https://github.com/ipfs/ipfs-desktop/releases/latest, please open a new issue with the output of `ls -la ~/.ipfs`.
